PokemonViewer/Model/PokemonSprite.swift

```diff
@@ -0,0 +1,36 @@
+//
+//  Pokemon.swift
+//  PokemonViewer
+//
+//  Created by Claire Chung on 2024/4/17.
+//
+
+import Foundation
+
+struct PokemonSprite: Hashable, Codable, Identifiable {
+    var id: Int
+    var name: String
+
+    struct TypeElement: Hashable, Codable {
+        struct TypeDetail: Hashable, Codable {
+            var name: String
+        }
+        var type: TypeDetail
+    }
+    var types: [TypeElement]
+    var typesNamesJoined: String {
+        types.map { $0.type.name }.joined(separator: ", ")
+    }
+
+    struct Sprites: Hashable, Codable, Equatable {
+        var imageUrl: URL?
+
+        enum CodingKeys: String, CodingKey {
+            case imageUrl = "front_default"
+        }
+    }
+    var sprites: Sprites
+    var imageUrl: URL? {
+        sprites.imageUrl
+    }
+}
```

Comment:
我會建議將從 JSON 轉換過來的參數跟經過運算轉換過的參數區分開來

可以利用 extension 的方式，或是純粹把相同類型的集中起來，然後用 `// MARK: something` 區分開來

```suggestion
struct PokemonSprite: Hashable, Codable, Identifiable {
    var id: Int
    var name: String
    var types: [TypeElement]
    var sprites: Sprites

    // MARK: - Computed properties    
    var typesNamesJoined: String {
        types.map { $0.type.name }.joined(separator: ", ")
    }
    
    var imageUrl: URL? {
        sprites.imageUrl
    }
}

// MARK: - Nested structures
extension PokemonSprite {
    struct TypeElement: Hashable, Codable {
        struct TypeDetail: Hashable, Codable {
            var name: String
        }
        var type: TypeDetail
    }
    
    struct Sprites: Hashable, Codable, Equatable {
        var imageUrl: URL?

        enum CodingKeys: String, CodingKey {
            case imageUrl = "front_default"
        }
    }
}
```